### PR TITLE
Check for missing typescript error when getting files by tsconfig

### LINKED ISSUE #1731: check for missing typescript error when getting files by tsconfig

### PR DESCRIPTION
**Pull request by saberduck**
Fixes #1731

### CODE REVIEW

File: eslint-bridge/src/tsconfig.ts [hunk 1]
```diff
@@ -46,5 +54,5 @@ export function getFilesForTsConfig(tsConfig: string): string[] {
   );
 
   // defensive empty array
```
Comment (vilchik-elena) [inline eslint-bridge/src/tsconfig.ts]: I don't get this comment, probably it's some leftover

File: eslint-bridge/tests/tsconfig.test.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+import { getFilesForTsConfig } from "../src/tsconfig";
+import { ParseExceptionCode } from "../src/parser";
+
+describe("tsconfig", () => {
+  it("should respond to JavaScript analysis request", () => {
```
Comment (vilchik-elena) [inline eslint-bridge/tests/tsconfig.test.ts]: is it copy-paste?

File: sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/eslint/AbstractEslintSensor.java [hunk 3]
```diff
@@ -105,6 +105,13 @@
       .map(checks::ruleKeyFor).orElse(null);
   }
 
+  static RuntimeException missingTypescript() {
+    LOG.error("TypeScript dependency was not found and it is required for analysis.");
+    LOG.error("Install TypeScript in the project directory or use NODE_PATH env. variable to set TypeScript " +
+      "location, if it's located outside of project directory.");
+    throw new IllegalStateException("Missing TypeScript dependency");
```
Comment (vilchik-elena) [inline sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/eslint/AbstractEslintSensor.java]: is it your intention to return RuntimeException but not doing it?

File: eslint-bridge/src/tsconfig.ts (hunk 1 above)
Comment (saberduck) [reply eslint-bridge/src/tsconfig.ts]: I was also not sure what does it mean :)

File: sonar-javascript-plugin/src/test/java/org/sonar/plugins/javascript/eslint/TsConfigFileTest.java [hunk 4]
```diff
@@ -36,14 +40,31 @@
...
+    assertThat(result).containsExactly(
+      entry("dir1/tsconfig.json", singletonList(inputFiles.get(0))),
+      entry("dir2/tsconfig.json", singletonList(inputFiles.get(1))),
+      entry("dir3/tsconfig.json", singletonList(inputFiles.get(2)))
+    );
+  }
+
   @Test
   public void failsToLoad() {
     EslintBridgeServer server = mock(EslintBridgeServer.class);
```
Comment (saberduck) [inline sonar-javascript-plugin/src/test/java/org/sonar/plugins/javascript/eslint/TsConfigFileTest.java]: this mock is not used anymore, I will remove it

File: sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/eslint/AbstractEslintSensor.java (hunk 3 above)
Comment (saberduck) [reply sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/eslint/AbstractEslintSensor.java]: no, there should be dedicated exception - I fixed it now

**Pull request merged**